src/cljs/rems/application.cljs

```diff
@@ -654,6 +656,7 @@
         invited-members (:application/invited-members application)
         permissions (:application/permissions application)
         can-add? (contains? permissions :application.command/add-member)
+        can-promote? (contains? permissions :application.command/promote-to-applicant)
         can-remove? (contains? permissions :application.command/remove-member)
         can-invite? (contains? permissions :application.command/invite-member)
         can-uninvite? (contains? permissions :application.command/uninvite-member)]
```

Comment:
This is starting to feel excessive, combined with the need to name and pass these forwards. Maybe we could have a higher order (container) component `with-permissions` that would add to props this kind of kw-parameters? Or use lower level (container) components that do roles checking?

Reply:
I tried to do a refactoring where we'd pass `permissions` directly to `member-info`, but unfortunately we have some special cases like the applicant (can't be removed) or invited members (can't be promoted even if promote-to-applicant is in permissions).

I guess I could push the handling of these special cases inside `member-info` as it already has the `invited-user?` parameter. In that case we could just pass `permissions` down.

Reply:
Yeah, I'll try this refactor out and push it if it feels good.

Reply:
The refactor came out pretty nicely I think!